backend/app/services/repositories/user_repository.py

```
@@ -0,0 +1,90 @@
+        query = self.db.query(User).filter(User.email == email)
+        if len(attributes) > 0:
+            query = query.options(load_only(*attributes))
+        elif not includePassword:
+            query = query.options(defer(User.password_hash, raiseload=True))
+
+        return query.first()
+
+    def get_user_by_id(
+        self,
+        user_id: int,
+        attributes: list[str] = [],
```

Ruff (0.14.10)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)